core/src/main/java/hudson/util/io/TarArchiver.java

```diff
@@ -82,21 +84,22 @@ public void visit(File file, String relativePath) throws IOException {
         if (Functions.isWindows())
             relativePath = relativePath.replace('\\', '/');
 
-        if (file.isDirectory())
+        BasicFileAttributes basicFileAttributes = Files.readAttributes(file.toPath(), BasicFileAttributes.class);
```

Comment:
Careful! `File#toPath` can throw `InvalidPathException`. If callers aren't already handling `InvalidPathException` (which is often the case for methods that rethrow `IOException` rather than catching and handling `IOException | InvalidPathException `), better to use `Util.fileToPath`.

Reply:
Thanks for the tip, adjusted as suggested.